src/FakeItEasy/Core/DynamicRaiser.cs

```diff
@@ -0,0 +1,42 @@
+namespace FakeItEasy.Core
+{
+    using System;
+    using System.Dynamic;
+#if FEATURE_NETCORE_REFLECTION
+    using System.Reflection;
+#endif
+    using FakeItEasy.Sdk;
+
+    internal class DynamicRaiser : DynamicObject, IEventRaiserArgumentProvider
+    {
+        private readonly object[] arguments;
+        private readonly EventHandlerArgumentProviderMap argumentProviderMap;
+
+        public DynamicRaiser(object[] arguments, EventHandlerArgumentProviderMap argumentProviderMap)
+        {
+            this.arguments = arguments;
+            this.argumentProviderMap = argumentProviderMap;
+        }
+
+        object[] IEventRaiserArgumentProvider.GetEventArguments(object fake)
+        {
+            return this.arguments;
+        }
+
+        public override bool TryConvert(ConvertBinder binder, out object result)
+        {
+            var targetType = binder.Type;
+            if (!targetType.GetTypeInfo().IsSubclassOf(typeof(Delegate)))
+            {
+                throw new InvalidCastException($"Unable to cast object of type '{this.GetType()}' to type '{targetType}'.");

```

Comment:
Ah. Good idea. In the end, it might be worthwhile commenting that to keep us from forgetting why the deviation.